snap/validate.go

```diff
@@ -270,12 +271,12 @@ func validateSocketAddr(socket *SocketInfo, fieldName string, address string) er
 
 func validateSocketAddrPath(socket *SocketInfo, fieldName string, path string) error {
 	if clean := filepath.Clean(path); clean != path {
-		return fmt.Errorf("socket %q has invalid %q: %q should be written as %q", socket.Name, fieldName, path, clean)
+		return fmt.Errorf("invalid %q: %q should be written as %q", fieldName, path, clean)
 	}
 
 	if !(strings.HasPrefix(path, "$SNAP_DATA/") || strings.HasPrefix(path, "$SNAP_COMMON/")) {
 		return fmt.Errorf(
-			"socket %q has invalid %q: only $SNAP_DATA and $SNAP_COMMON prefixes are allowed", socket.Name, fieldName)
+			"invalid %q: only $SNAP_DATA and $SNAP_COMMON prefixes are allowed", fieldName)
```

Comment:
I've got a problem with this one: `only FOO and BAR prefixes are allowed` does not mean quite the same as `must have a prefix of FOO or BAR`, which is the case here. Granted this one was wrong before :-)